pallets/withdraw-teleport/src/lib.rs

```diff
@@ -101,7 +102,28 @@ pub mod pallet {
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		#[pallet::call_index(0)]
-		#[pallet::weight(<pallet_xcm::TestWeightInfo as pallet_xcm::WeightInfo>::teleport_assets())]
+		#[pallet::weight({
+			let native_asset = MultiAsset {
+				id: AssetId::Concrete(MultiLocation::here()),
+				fun: Fungibility::Fungible(*native_asset_amount),
+			};
+			let native_assets = MultiAssets::from(vec![native_asset.clone()]);
+			let maybe_assets: Result<MultiAssets, ()> = (*fee_asset.clone()).try_into();
+			match maybe_assets {
+				Ok(assets) => {
+					use sp_std::vec;
+					let mut message = Xcm(vec![
+						WithdrawAsset(native_assets.clone()),
+						SetFeesMode { jit_withdraw: true },
+						BurnAsset(native_assets),
+						WithdrawAsset(assets.clone()),
+						BurnAsset(assets),
+					]);
+					T::Weigher::weight(&mut message).map_or(Weight::MAX, |w| <T as pallet::Config>::WeightInfo::withdraw_and_teleport().saturating_add(w))
```

Comment:
I'm not 100% sure about this but here is my concern:

You are computing the weight for the XCM messages and once you have the result you sum that weight to the benchmarking for the extrinsic weight.  Aren't these XCM messages charged already on the XCM Executor when the messages are processed? In other terms , are you not charging twice the XCM messages?

Reply:
I stuck to the pattern used on [pallet-xcm](https://github.com/paritytech/polkadot/blob/52209dcfe546ff39cc031b92d64e787e7e8264d4/xcm/pallet-xcm/src/lib.rs#L806) and traced relevant [PR](https://github.com/paritytech/polkadot/issues/7182) even though I agree it's not fully clear to me either. Already discussed this with xcm team and agreed on keeping it until they address it further just to be sure on not being undercharging. 
What I do also realize is that if I want to fully stick to the pattern I should also add `pallet-xcm` `send` weight as it is also called inside the extrinsic, this is opaque on the case of `teleport` and `reserve` as it is done inside the instructions.

Reply:
What I do also realize is that if I want to fully stick to the pattern I should also add `pallet-xcm` `send` weight as it is also called inside the extrinsic, this is opaque on the case of teleport and reserve as it is done inside the instructions.